### PR TITLE
Enable environment support for Choreo - Moesif integration

### PR DESCRIPTION
**Pull request by Lasith650**
Related with: https://github.com/wso2-enterprise/choreo/issues/22229

### CODE REVIEW

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java [hunk 1]
```diff
@@ -82,15 +82,28 @@ private void doRetry(String orgId, MetricEventBuilder builder) {
...
+            if (orgIdEnvMap.containsKey(orgId)) {
+                userSelectedEnvironment = orgIdEnvMap.get(orgId);
+            } else {
+                return;
+            }
         } else {
             return;
         }
 
+        if (userSelectedEnvironment.equals(Constants.PRODUCTION) && !eventEnvironment.equals(Constants.PRODUCTION)) {
```
Comment (jaadds) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java:103]: Do the equal comparison against the constant, i.e Constants.PRODUCTION.equals(userSelectedEnvironment). This prevents NPEs.

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java [hunk 2]
```diff
@@ -82,15 +82,28 @@ private void doRetry(String orgId, MetricEventBuilder builder) {
     public void publish(MetricEventBuilder builder) throws MetricReportingException {
         Map<String, Object> event = builder.build();
         ConcurrentHashMap<String, String> orgIDMoesifKeyMap = keyRetriever.getMoesifKeyMap();
+        ConcurrentHashMap<String, String> orgIdEnvMap = keyRetriever.getEnvMap();
+        LinkedHashMap properties = (LinkedHashMap) event.get("properties");
```
Comment (jaadds) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java:86]: Make this a constant.

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java (hunk 1 above)
Comment (Lasith650) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java]: Resolved with: 9a0b30dc39ce6c0986d843daf6803adebbe92e56

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java (hunk 2 above)
Comment (Lasith650) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/MoesifClient.java]: Resolved with: 9a0b30dc39ce6c0986d843daf6803adebbe92e56

**Review by jaadds — APPROVED** [review]

**Pull request merged**